### PR TITLE
add register impl

### LINKED ISSUE #1: Set up user API
possibly don't need platform specific stuff here, the only thing that really requires that is the base API class, since it will need to handle platform specific coroutines.

### PR DESCRIPTION
**Pull request by eiknat**

### CODE REVIEW

File: src/commonMain/kotlin/com/eiknat/lemmyclient/api/API.kt [hunk 1]
```diff
@@ -2,9 +2,13 @@ package com.eiknat.lemmyclient.api
...
 import com.eiknat.lemmyclient.APIResult
 
-expect abstract class API {
+expect open class API() {
+
+	suspend fun <T> performRequest(op: RequestOp): APIResult<T>
 
     suspend fun performRequestAsync(op: RequestOp): APIResult<ResponseOp>
 
+    @Suppress("UNCHECKED_CAST")
```
Comment (5HT2) [inline src/commonMain/kotlin/com/eiknat/lemmyclient/api/API.kt:11]: Why are we mixing tabs and spaces?

File: src/commonMain/kotlin/com/eiknat/lemmyclient/apipublic/user/Register.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+	val email: String?
+	val password: String
+	val passwordVerification: String
+	val admin: Boolean
+	val showNsfw: Boolean
+}
+
+interface RegisterResponse {
+	val jwt: String
+}
```
Comment (5HT2) [inline src/commonMain/kotlin/com/eiknat/lemmyclient/apipublic/user/Register.kt:15]: Some classes are using tabs, some spaces. I feel this should be more consistent.

**Review by 5HT2 — CHANGES_REQUESTED** [review]

File: src/commonMain/kotlin/com/eiknat/lemmyclient/apipublic/user/Register.kt (hunk 2 above)
Comment (eiknat) [reply src/commonMain/kotlin/com/eiknat/lemmyclient/apipublic/user/Register.kt:15]: my ide broke ._.

**Comment by eiknat** [comment]
@dominikaaaa does this make sense to you? i feel like there's a better way to do this but i just can't picture it

**Comment by 5HT2** [comment]
Makes sense. I'm not sure if there's a l
cleaner way

**Pull request merged**